cdci_data_analysis/analysis/instrument.py

```diff
@@ -551,7 +551,8 @@ def set_catalog_from_fronted(self,par_dic,request,back_end_query,logger=None,ver
             #print('POST')
             try:
                 cat_file_path = back_end_query.upload_file('user_catalog_file', back_end_query.scratch_dir)
-                par_dic['user_catalog_file'] = cat_file_path
+                if cat_file_path is not None:
```

Comment:
could there be a test? Cat uploading caused quite some problems, it should be tested. If you do not want, let's make an issue and we do soon after.

Reply:
I do not fully know this catalog functionality, and not aware of other issues. So anyway will need some clarifications to write meaningful tests. Thus better to split into another PR.

Reply:
ok #173